tonic/src/status.rs

```diff
@@ -800,10 +820,18 @@ mod tests {
     #[test]
     #[cfg(feature = "transport")]
     fn from_error_h2() {
+        use std::error::Error as _;
+
         let orig = h2::Error::from(h2::Reason::CANCEL);
         let found = Status::from_error(&orig);
 
         assert_eq!(found.code(), Code::Cancelled);
+
+        let source = found
+            .source()
+            .and_then(|err| err.downcast_ref::<h2::Error>())
+            .unwrap();
+        assert_eq!(orig.reason(), source.reason());
```

Comment:
I'm thinking it would also be nice to have a high level integration test kinda thing that uses the APIs users would. Not sure how to write such a test though.

Reply:
The `h2` crate has test helpers for low level testing of HTTP/2 streams. Maybe this integration test would be similar to the tests in https://github.com/hyperium/h2/tree/master/tests/h2-tests?

Reply:
Any thoughts on whether an h2-level integration test would be the way to go?

Reply:
I don't think we need those low level helpers, we should be able to trigger most things from hyper/h2 directly, no?